src/frontend/handlers.go

```diff
@@ -189,21 +189,33 @@ func (fe *frontendServer) productHandler(w http.ResponseWriter, r *http.Request)
 		Price *pb.Money
 	}{p, price}
 
+	// Fetch packaging info (weight/dimensions) of the product
+	// The packaging service is an optional microservice you can run as part of a Duet AI demo.
+	packagingInfo := PackagingInfo{}
+	if isPackagingServiceConfigured() {
+		packagingInfo, err = httpGetPackagingInfo(id)
+		if err != nil {
+			fmt.Println("Failed to obtain product's packaging info:", err)
+		}
+	}
+
 	if err := templates.ExecuteTemplate(w, "product", map[string]interface{}{
-		"session_id":        sessionID(r),
-		"request_id":        r.Context().Value(ctxKeyRequestID{}),
-		"ad":                fe.chooseAd(r.Context(), p.Categories, log),
-		"user_currency":     currentCurrency(r),
-		"show_currency":     true,
-		"currencies":        currencies,
-		"product":           product,
-		"recommendations":   recommendations,
-		"cart_size":         cartSize(cart),
-		"platform_css":      plat.css,
-		"platform_name":     plat.provider,
-		"is_cymbal_brand":   isCymbalBrand,
-		"deploymentDetails": deploymentDetailsMap,
-		"frontendMessage":   frontendMessage,
+		"session_id":                   sessionID(r),
+		"request_id":                   r.Context().Value(ctxKeyRequestID{}),
+		"ad":                           fe.chooseAd(r.Context(), p.Categories, log),
+		"user_currency":                currentCurrency(r),
+		"show_currency":                true,
+		"currencies":                   currencies,
+		"product":                      product,
+		"recommendations":              recommendations,
+		"cart_size":                    cartSize(cart),
+		"platform_css":                 plat.css,
+		"platform_name":                plat.provider,
+		"is_cymbal_brand":              isCymbalBrand,
+		"deploymentDetails":            deploymentDetailsMap,
+		"frontendMessage":              frontendMessage,
+		"packagingInfo":                packagingInfo,
+		"isPackagingServiceConfigured": isPackagingServiceConfigured(),
```

Comment:
Sorry, Golang's auto-formatting confused this diff:
All I'm adding:
```
		"packagingInfo":                packagingInfo,
		"isPackagingServiceConfigured": isPackagingServiceConfigured(),
```

Reply:
Is `isPackagingServiceConfigured` necessary if e.g. `packagingInfo` would be null or empty if it's not configured (thus you could do `if packageInfo != null` for example)? Or can `packagingInfo` be null or empty even if packaging is configured / enabled configured?

Reply:
Good question.

I've updated the code the use _**pointers**_ to the PackagingInfo struct:
```
var packagingInfo *PackagingInfo = nil
```
and
```
func httpGetPackagingInfo(productId string) (*PackagingInfo, error) {
```

The HTML template no longer uses a `isPackagingServiceConfigured` boolean.